### PR TITLE
XIVY-12449 add first electron playwright tests & integration build

### PR DESCRIPTION
**Pull request by ivy-lmu**
new build: https://jenkins.ivyteam.io/job/vscode-extensions_integration/

### CODE REVIEW

File: build/integration/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM mcr.microsoft.com/playwright:next
```
Comment (ivy-cst) [inline build/integration/Dockerfile:1]: I would use the same version as in yarn.lock

**Review by ivy-cst — APPROVED** [review]
Very nice :+1:. May be we should include a real test project?

File: playwright/package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "private": true,
+  "name": "@axonivy/vscode-extensions-playwright",
+  "version": "12.2.0",
```
Comment (ivy-lli) [inline playwright/package.json:4]: 12.2.0?

File: playwright/tests/page-objects/view.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { PageObject } from './page-object';
+
+export interface ViewData {
+  tabSelector: string;
+  viewSelector: string;
+  viewName?: string;
+}
+
+export class View extends PageObject {
+  constructor(private data: ViewData, page: Page) {
```
Comment (ivy-lli) [inline playwright/tests/page-objects/view.ts:11]: i would also mark this parameters as readonly to make them final.

File: playwright/tests/engine.test.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  test('check if embedded engine has started', async ({ window }) => {
+    const outputview = new OutputView(window);
+    await outputview.isTabVisible();
+    await outputview.isActive();
+    await outputview.isViewVisible();
+    await outputview.checkIfEngineStarted();
+  });
+
+  test('check default engine settings', async ({}, testInfo) => {
+    const app = await launchElectronApp(noProjectWorkspacePath, testInfo.title);
```
Comment (ivy-lli) [inline playwright/tests/engine.test.ts:19]: not sure about this own test extension...
Doesn't start this test now the ElectronApp twice? Once in the test and once here?

But I assume that you need it right? So playwright doesn't simply start a browser?

But maybe we can add some custom parameters to our own test method so we can directly add the workspace there so it is correctly started right away?

**Review by ivy-lli — APPROVED** [review]
Looks good to me in general, however I have some suggestions (which I don't know if everything is possible :) ):

File: playwright/package.json (hunk 2 above)
Comment (ivy-lmu) [reply playwright/package.json:4]: ups :D

**Comment by ivy-lmu** [comment]
> Very nice 👍. May be we should include a real test project?

I will add one at the time it is needed

File: playwright/tests/engine.test.ts (hunk 4 above)
Comment (ivy-lmu) [reply playwright/tests/engine.test.ts:19]: > Doesn't start this test now the ElectronApp twice? Once in the test and once here?

no, if `{ window }`is not passed, the test extension does not run. So, it's started once here.

yes, browser does not work :(

> But maybe we can add some custom parameters to our own test method so we can directly add the workspace there so it is correctly started right away?

I tried to do that, but could not find a good solution yet. But I plan to improve it somehow in a followup

**Pull request merged**